backend/src/graphql_api/plt_event_metrics.sql

```diff
@@ -0,0 +1,110 @@
+-- Inputs:
+-- $1::interval - total period
+-- $2::interval - bucket width 
+-- $3::text     - optional token_id (NULL means include all tokens)
+-- This is temprary, might have to modify later
+WITH bucket_time AS (
+  SELECT generate_series(
+    now() - $1::interval,
+    now(),
+    $2::interval
+  ) AS bucket_start
+),
+
+events_with_block AS (
+  SELECT
+    e.*,
+    pt.token_id,
+    b.slot_time
+  FROM plt_events e
+  JOIN plt_tokens pt ON pt.index = e.token_index
+  JOIN transactions tx ON e.transaction_index = tx.index
+  JOIN blocks b ON tx.block_height = b.height
+),
+
+supply_with_block AS (
+  SELECT
+    pt.token_id,
+    b.slot_time,
+    (pt.total_minted - pt.total_burned) / POWER(10, pt.decimal) AS total_supply
+  FROM plt_tokens pt
+  JOIN transactions tx ON pt.transaction_index = tx.index
+  JOIN blocks b ON tx.block_height = b.height
+),
+
+holders_with_block AS (
+  SELECT
+    paa.account_index,
+    pt.token_id,
+    paa.amount,
+    b.slot_time
+  FROM plt_accounts paa
+  JOIN plt_tokens pt ON paa.token_index = pt.index
+  JOIN transactions tx ON pt.transaction_index = tx.index
+  JOIN blocks b ON tx.block_height = b.height
+  WHERE paa.amount > 0
+),
+
+aggregated AS (
+  SELECT
+    bt.bucket_start,
+
+    -- Event counts
+    (
+      SELECT COUNT(*) FROM events_with_block e
```

Comment:
I don't think that `COUNT(*)` will perform well once we have lot's of data in the database (and large bucket sizes e.g. 1 year). Would be good to test this metrics against a large dataset in the database and analyze its performance to ensure we don't need to re-write it in the future.

Reply:
Have we done some testing on large datasets?